src/shared/Microsoft.Git.CredentialManager/IGit.cs

```diff
@@ -13,13 +13,56 @@ public interface IGitConfiguration : IDisposable
         /// <param name="cb">Callback to invoke for each configuration entry.</param>
         void Enumerate(GitConfigurationEnumerationCallback cb);
 
+        /// <summary>
+        /// Get a snapshot of the configuration filtered to the specified level.
+        /// </summary>
+        /// <param name="level">Configuration level filter.</param>
+        /// <returns>Git configuration snapshot.</returns>
+        IGitConfiguration GetFilteredConfiguration(GitConfigurationLevel level);
+
         /// <summary>
         /// Try and get the value of a configuration entry as a string.
         /// </summary>
         /// <param name="name">Configuration entry name.</param>
         /// <param name="value">Configuration entry value.</param>
         /// <returns>True if the value was found, false otherwise.</returns>
         bool TryGetValue(string name, out string value);
+
+        /// <summary>
+        /// Set the value of a configuration entry.
+        /// </summary>
+        /// <param name="name">Configuration entry name.</param>
+        /// <param name="value">Configuration entry value.</param>
+        void SetValue(string name, string value);
+
+        /// <summary>
+        /// Deletes a configuration entry from the highest level.
+        /// </summary>
+        /// <param name="name">Configuration entry name.</param>
+        void DeleteEntry(string name);
+
+        /// <summary>
+        /// Get all values of a multivar configuration entry.
+        /// </summary>
+        /// <param name="name">Configuration entry name.</param>
+        /// <param name="regexp">Regular expression to filter which variables we're interested in. Use null to indicate all.</param>
+        /// <returns>All values of the multivar configuration entry.</returns>
+        IEnumerable<string> GetMultivarValue(string name, string regexp);
+
+        /// <summary>
+        /// Set a multivar configuration entry value.
+        /// </summary>
+        /// <param name="name">Configuration entry name.</param>
+        /// <param name="regexp">Regular expression to indicate which values to replace.</param>
+        /// <param name="value">Configuration entry value.</param>
+        void SetMultivarValue(string name, string regexp, string value);
```

Comment:
It's unclear to me from the interface docs if this appends or replaces values. Can we clarify?

Reply:
It can do either, depending on the value of the regular expression.

The [libgit2 docs](https://libgit2.org/libgit2/#HEAD/group/config/git_config_set_multivar) for the underlying method `git_config_set_multivar` is:

> Set a multivar in the local config file.
>
> git_config * cfg : where to look for the variable
> const char * name : the variable's name
> const char * regexp : a regular expression to indicate which values to replace
> const char * value : the new value.

What I've observe happens (in pseudo code) is:

```python
for entry in entries
    if entry.name matches regexp
        entry.value = value
        return

newEntry := { name, value }
entries.add(newEntry)
```

..in a sense "setting/replacing" the value of the 'matched entry' (and if that is an entry that doesn't exist, it is created and the value 'set/replaced').

Reply:
I've added a `<remarks/>` section indicating this behaviour.